File: problems/0347-top-k-frequent-elements/python/solution.py
def topKFrequent(nums, k):
    """Top K Frequent Elements — https://leetcode.com/problems/top-k-frequent-elements/"""
    counts = {}
    for n in nums:
        counts[n] = counts.get(n, 0) + 1
    buckets = [[] for _ in range(len(nums) + 1)]
    for n, c in counts.items():
        buckets[c].append(n)
    res = []
    for c in range(len(buckets) - 1, 0, -1):
        for n in buckets[c]:
            res.append(n)
            if len(res) == k:
                return res
    return res

File: problems/0347-top-k-frequent-elements/go/solution.go
package solution

// Top K Frequent Elements — https://leetcode.com/problems/top-k-frequent-elements/
func topKFrequent(nums []int, k int) []int {
	counts := make(map[int]int)
	for _, n := range nums {
		counts[n]++
	}
	buckets := make([][]int, len(nums)+1)
	for n, c := range counts {
		buckets[c] = append(buckets[c], n)
	}
	res := make([]int, 0, k)
	for c := len(buckets) - 1; c > 0 && len(res) < k; c-- {
		for _, n := range buckets[c] {
			res = append(res, n)
			if len(res) == k {
				break
			}
		}
	}
	return res
}
